src/foam/core/reflow/Mapping.js

```diff
@@ -4,35 +4,207 @@
  * http://www.apache.org/licenses/LICENSE-2.0
  */
 
+foam.ENUM({
+  package: 'foam.core.reflow',
+  name: 'MappingType',
+
+  values: [
+    {
+      name: 'CONSTANT',
+      label: 'Constant',
+      documentation: 'Static value that is applied to all rows'
+    },
+    {
+      name: 'FIELD',
+      label: 'Field',
+      documentation: 'Value comes from a field/column in the input data'
+    },
+    {
+      name: 'DYNAMIC',
+      label: 'Dynamic',
+      documentation: 'Value computed from JavaScript expression'
```

Comment:
If we want to run the same mappings on the server for SFTP then this won't work. We should use FScript instead which we can run in both browser and server.